Content.Server/GameObjects/Components/Buckle/BuckleComponent.cs

```diff
@@ -427,15 +430,33 @@ public override ComponentState GetComponentState()
             {
                 drawDepth = BuckledTo.SpriteComponent.DrawDepth - 1;
             }
+            
 
-            return new BuckleComponentState(Buckled, drawDepth);
+            return new BuckleComponentState(Buckled, drawDepth, EntityBuckledTo, DontCollide);
         }
 
         bool IInteractHand.InteractHand(InteractHandEventArgs eventArgs)
         {
             return TryUnbuckle(eventArgs.User);
         }
 
+
+        public void Update()
+        {
+            if (!DontCollide || Body == null)
+                return;
+
+            Body.WakeBody();
+
+            if (!IsOnStrapEntityThisFrame && DontCollide)
+            {
+                DontCollide = false;
+                Dirty();
+            }
```

Comment:
Set the buckledto entity to null as well here.

Reply:
I've added TryUnbuckle(Owner); to the if code block to set BuckledTo to null and update everything associated to it.